lock-impl/src/main/java/com/palantir/lock/impl/LegacyTimelockService.java

```diff
@@ -107,15 +109,17 @@ public LockImmutableTimestampResponse lockImmutableTimestamp() {
     }
 
     @Override
-    public StartIdentifiedAtlasDbTransactionResponse startIdentifiedAtlasDbTransaction() {
-        LockImmutableTimestampResponse immutableTimestamp = lockImmutableTimestamp();
+    public List<StartIdentifiedAtlasDbTransactionResponse> startIdentifiedAtlasDbTransactionBatch(int count) {
+        List<StartIdentifiedAtlasDbTransactionResponse> responses = new ArrayList<>();
         try {
-            return StartIdentifiedAtlasDbTransactionResponse.of(
-                    immutableTimestamp,
-                    TimestampAndPartition.of(getFreshTimestamp(), 0));
+            IntStream.range(0, count).forEach(
+                    $ -> responses.add(StartIdentifiedAtlasDbTransactionResponse.of(lockImmutableTimestamp(),
+                            TimestampAndPartition.of(getFreshTimestamp(), 0))));
+            return responses;
         } catch (RuntimeException | Error throwable) {
             try {
-                unlock(ImmutableSet.of(immutableTimestamp.getLock()));
+                unlock(responses.stream().map(response -> response.immutableTimestamp().getLock()).collect(
+                        Collectors.toSet()));
```

Comment:
Again, this is now a single call, so there is no handling for what happens if there is a failure half-way through closing, as such a failure cannot occur.